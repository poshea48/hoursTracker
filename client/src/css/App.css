/* #root {
  min-height: 100vh;
} */
.App {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.app-content {
  flex: 1 auto;
}

.container {

}

a {
  text-decoration: none;
}

.landing {
  position: relative;
  /* background: url('./img/showcase.jpg') no-repeat; */
  background-size: cover;
  background-position: center;
  height: 100vh;
  margin-top: -24px;
  margin-bottom: -50px;
}

.landing-inner {
  padding-top: 80px;
}

.dark-overlay {
  /* background-color: rgba(0, 0, 0, 0.7); */
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.landing-links {
  /* justify-content: space-around !important; */
  margin: 0 !important;
}
.main-title {
  flex-direction: column !important;
}
.row-item {
  text-align: center;
}

.main {
  flex: 1;
}

.navbar {
  display: flex;
  background-color: #343a40;
  color: #fff;
  justify-content: space-between;
}

.auth-button {
  cursor: pointer;
  border-radius: 5px;
  color: #fff;
  background: #343a40;
}

.footer {
  height: 50px;
  background-color: #343a40;
  margin-top: 50px;
}

.nav-item {
  width: 20%;
  align-self: center;
  margin: 15px 0;
  margin-left: 1em;
}

.auth {
  width: 30%;
}

.auth ul {
  display: flex;
  list-style: none;
  justify-content: flex-end;
  padding: 0;
  margin: 0;
}

.auth li {
  margin-right: 2em;
}
.header {
  justify-content: center;
}

.history-layout {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.row {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin: 10px auto;
}

.history-layout:first-child {
  border-right: 1px solid #afafac;

}

.compare {
  text-align: center;
}

.heading {
  color: rgba(7,27,82,1)
}

.actions {
  display: flex;
  justify-content: space-evenly;
}

.button {
  cursor: pointer;
  /* background-color: #42f4c2; */
  background-color: #06db6d;
  /* background-color: #3befc9; */
  border: 1px solid black;
  width: 10em;
  padding: .6em;
  border-radius: 10px;
  transition: all .3s ease-in-out;
  margin: .2em 0;
}

.button:hover {
  transform: scale(1.1);
  opacity: 2;
  font: 2em;
}

.button:disabled {
  background: #e7040f;
  border: 1px solid #e7040f;
  opacity: .6;
  transition: none;
  transform: none;
  cursor: default;
}

.history {
  margin: 20px 2em;
}

.chart {
  display: flex;
  justify-content: center;
  margin-bottom: 20px;
}

.chart-content {
  position: relative;
  max-width: 800px;
  height: 200px;
  background: #ddd;
}

@media only screen and (max-width: 670px) {
  .actions {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    align-items: center;
  }

  .button {
    width: 8em;
    margin: 5px;
  }

}

@media only screen and (max-width: 560px) {
  .actions {
    justify-content: center;
  }

  .button {
    margin: 2px;
    width: 6em;
  }

  .auth {
    width: 70%;
  }

}
