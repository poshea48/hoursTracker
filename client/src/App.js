import React, { Component } from "react";
import { Route, Switch } from "react-router-dom";
import jwt_decode from "jwt-decode";
import setAuthToken from "./utils/setAuthToken";
import { setCurrentUser, logoutUser } from "./redux/actions/authActions";
import { logHours } from "./redux/actions/timerActions";
import { Provider } from "react-redux";
import store from "./redux/store";

import "./css/App.css";
import MainPage from "./layout/MainPage";
import Landing from "./layout/Landing";
import Login from "./auth/Login";
import Register from "./auth/Register";
import PrivateRoute from "./common/PrivateRoute";
import Footer from "./layout/Footer";

// Check for token
if (localStorage.jwtTokenHoursTracker) {
  setAuthToken(localStorage.jwtTokenHoursTracker);
  const decoded = jwt_decode(localStorage.jwtTokenHoursTracker);
  store.dispatch(setCurrentUser(decoded));

  // Check for expired token
  const currentTime = Date.now() / 1000;
  if (decoded.exp < currentTime) {
    const { hoursToday, dateToday } = store.getState().timer;

    if (hoursToday > 0) {
      store.dispatch(logHours(hoursToday, dateToday));
    }
    store.dispatch(logoutUser());
    window.location.href = "/login";
  }
}

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className="App">
          <div className="app-content">
            <Route exact path="/" component={Landing} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/register" component={Register} />
            <Switch>
              <PrivateRoute exact path="/dashboard" component={MainPage} />
            </Switch>
          </div>
          <Footer />
        </div>
      </Provider>
    );
  }
}

export default App;
