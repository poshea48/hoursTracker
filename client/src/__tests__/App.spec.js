import React from "react";
import { shallow, configure } from "enzyme";
import App from "../App";

describe("App", () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<App />)));

  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a div", () => {
    expect(wrapper.find("div").length).toEqual(0);
  });
});
