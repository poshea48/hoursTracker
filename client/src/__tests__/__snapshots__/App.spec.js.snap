// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/login"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/register"
        />,
        <Switch>
          <Connect(PrivateRoute)
            component={[Function]}
            exact={true}
            path="/dashboard"
          />
        </Switch>,
      ],
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/register",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(PrivateRoute)
            component={[Function]}
            exact={true}
            path="/dashboard"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/dashboard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/register"
          />,
          <Switch>
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/dashboard"
            />
          </Switch>,
        ],
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/dashboard"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
